tuist/tuist
# Improve default swift version handling

Resolves: https://github.com/tuist/tuist/issues/4672

### Short description 📝

- The default `SWIFT_VERSION` build setting was inferred from the system swift version
- Xcode is mostly ok with that, however the values aren't correct
  - Xcode currently expects one of 3 values for this build setting, `4.0`, `4.2` and `5.0`
- The system Swift version includes minor and patch versions (e.g. `5.6.1`)
- To ensure the correct version is used, we no longer attempt to infer the swift version from the system, rather have a staticly hard coded one
  - Users are still able to override this in either `Project.settings` or `Target.settings`
- Additionally the documentation around `Config.swiftVersion` was updated to clarify it isn't used to control the `SWIFT_VERSION` build setting


### How to test the changes locally 🧐

- Generate any fixture / project
- Verify the `SWIFT_VERSION` build setting is `5.0`

### Checklist ✅
